lib/Support/CompilationLog.cpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Support/CompilationLog.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include <fstream>
+
+namespace glow {
+static llvm::cl::opt<bool>
+    dumpCompilationLogOpt("dump-compilation-log",
+                          llvm::cl::desc("Dump compilation log"));
+
+std::vector<std::string> CompilationScope::scopeStack_ = {};
+
+CompilationScope::CompilationScope(std::string scopeName)
+    : scopeName_(scopeName) {
+  scopeStack_.push_back(scopeName);
+  if (dumpCompilationLogOpt) {
+    dumpCompilationLog(this->getCurrentCompilationScopeStack());
+  }
+}
+
+CompilationScope::~CompilationScope() { scopeStack_.pop_back(); }
+
+llvm::StringRef CompilationScope::getCurrentCompilationScopeStack() {
+  std::string scopes = "";
+  for (auto scopeStr : scopeStack_) {
+    scopes = scopes + "-->" + scopeStr;
```

Comment:
Consider using `std::string::append` or `std::string::operator+=` here instead of `std::string::operator+` to avoid a a few extra copies and assignment.